### PR TITLE
complete the gentoo init

### PR DESCRIPTION
**Pull request by acidwars**
This works for me...

### CODE REVIEW

**Comment by Godzil** [comment]
The _frabjous_ overlay have a really simple and working well init script, why not using this one?

```
#!/sbin/openrc-run
# Copyright 1999-2017 Gentoo Foundation
# Distributed under the terms of the GNU General Public License v2

description="Gogs, a self-hosted Git service"

: ${GOGS_USER:=git}
: ${GOGS_GROUP:=git}
: ${GOGS_LOG:="/var/log/${SVCNAME}/${SVCNAME}.log"}
: ${GOGS_OPTS:="--config /var/lib/gogs/conf/app.ini"}

command="/usr/bin/gogs"
command_args="web ${GOGS_OPTS}"
command_background=yes
pidfile="/run/${SVCNAME}/${SVCNAME}.pid"
start_stop_daemon_args="-u ${GOGS_USER}:${GOGS_GROUP} \
	-e GOGS_CUSTOM=/var/lib/gogs -e USER=${GOGS_USER} \
	-1 ${GOGS_LOG} -2 ${GOGS_LOG}"

depend() {
	need localmount net
}

start_pre() {
	checkpath -d -m 0755 -o "${GOGS_USER}":"${GOGS_GROUP}" "${pidfile%/*}"
}

```

File: scripts/init/gentoo/gogs [hunk 1]
```diff
@@ -1,16 +1,48 @@
...
 
-depend()
-{
-    need net
+stop() {
+        ebegin "Stopping Gogs"
+        RETVAL="$?"
+        [ "$RETVAL" = 2 ] && return 2
+        start-stop-daemon --stop --quiet --oknodo --pidfile $PIDFILE
+        [ "$?" = 2 ] && return 2
```
Comment (csmk) [inline scripts/init/gentoo/gogs]: Those tests are redundant, as Gogs doesn't handle pidfile by itself.

File: scripts/init/gentoo/gogs [hunk 2]
```diff
@@ -1,16 +1,48 @@
...
+                --test --chdir $WORKINGDIR --chuid $USER \\
+                --exec $DAEMON -- $DAEMON_ARGS > /dev/null \\
+                || return 1"
+        sh -c "USER=$USER start-stop-daemon --start --quiet --pidfile $PIDFILE --make-pidfile \\
+                --background --chdir $WORKINGDIR --chuid $USER \\
+                --exec $DAEMON -- $DAEMON_ARGS \\
+                || return 2"
+        eend $?
 
-start_stop_daemon_args="--user ${USER} --chdir ${DIR}"
```
Comment (csmk) [inline scripts/init/gentoo/gogs]: These variables (command, command_args, pidfile and so on) are special ones. You shouldn't drop them in favor of writing the same (redundant) thing in start() and stop().

**Comment by Unknwon** [comment]
Hmm... @acidwars  any updates to the review comments made by @csmk?

**Comment by csmk** [comment]
@acidwars please, consider this revision I made based on the one distributed in my overlay.

```sh
#!/sbin/openrc-run
# Copyright 1999-2017 Gentoo Foundation
# Distributed under the terms of the GNU General Public License v2

description="Gogs, a self-hosted Git service"

: ${GOGS_BIN:="$(which gogs)"}
: ${GOGS_USER:=gogs}
: ${GOGS_HOME:="/var/lib/gogs"}
: ${GOGS_LOG:="/var/log/gogs/${RC_SVCNAME}.log"}
: ${GOGS_CONFIG:="${GOGS_HOME}/conf/app.ini"}
: ${SSD_OPTS:="--wait 1000 -1 ${GOGS_LOG} -2 ${GOGS_LOG}"}

name="${GOGS_NAME:-"Gogs"}"
command="${GOGS_BIN}"
command_args="web --config ${GOGS_CONFIG} ${GOGS_OPTS}"
command_user="${GOGS_USER}"
command_background=true
pidfile="/run/${RC_SVCNAME}.pid"
required_files="${GOGS_CONFIG}"
retry=${GOGS_TERMTIMEOUT:-"TERM/60/KILL/5"}
start_stop_daemon_args="${SSD_OPTS} \
	-e \"GOGS_CUSTOM=${GOGS_HOME}\" \
	-e \"USER=${GOGS_USER}\""

depend() {
	need localmount net
}

start_pre() {
	checkpath -d -m 0700 -o ${GOGS_USER} "${GOGS_HOME}" "${GOGS_LOG%/*}"
	checkpath -f -m 0600 -o ${GOGS_USER} "${GOGS_CONFIG}"
}
```

As you can see, instead of using "hardcoded" variables, the user can easily changes it by just creating a file using the same name from the init script, which must be located in `/etc/conf.d` (e.g. `/etc/conf.d/gogs`). This way we can keep the init script untouched.

**Note:** If the user wants to define a group that is not the primary one, he/she just needs to add a colon between username and group in the variable `GOGS_USER` (e.g. `myuser:myanothergroup`).

**Comment by csmk** [comment]
I also suggest to change the name of folder "gentoo" in Gogs's repo, to "openrc", because the init script above can also be used by other OpenRC-based distros, like Alpine Linux and Artix.

**Comment by acidwars** [comment]
As I am not using Gogs on Gentoo anymore, maybe there should be a new pull request with the change from @csmk

**Pull request closed without merging**